lib/index.js

```diff
@@ -256,10 +256,20 @@ function __(StreamCtor) {
             var mapper = hintMapper(mappingHint);
 
             s = new StreamCtor();
-            secondArg.on(xs, function () {
+
+            var callback_func = function () {
                 var ctx = mapper.apply(this, arguments);
                 s.write(ctx);
-            });
+            };
+
+            secondArg.on(xs, callback_func);
+            var removeMethod = secondArg.removeListener || secondArg.unbind;
```

Comment:
Can you add a comment that says which method belongs to which type of event emitters (EventEmitter or jQuery)?